Lombiq.Tests.UI/Services/UITestExecutionSession.cs

```diff
@@ -94,6 +98,13 @@ public async Task<bool> ExecuteAsync(int retryCount, string dumpRootPath)
 
                 await SetupAsync();
             }
+            else
+            {
+                if (_setupSnapshotDirectoryExists)
```

Comment:
Add a comment about when it can happen that there's a snapshot but not a setup operation.